### PR TITLE
Add a retroactive approver address for deploy bypass

### PR DESCRIPTION
**Pull request by apanzerj**
Reviewers: @pswadi-zendesk @gabetax @jonmoter 

Description:

Adding one extra CC, the person who will do the retroactive approval when the bypass deploy is utilized.

cc: @pdeuter

### CODE REVIEW

**Comment by pswadi-zendesk** [comment]
Since this if or audit purpose, could we reuse audit email we already have? 
https://github.com/zendesk/samson/blob/master/.env.example#L28

**Comment by apanzerj** [comment]
Is that being used inside Zendesk at the moment?

**Comment by pswadi-zendesk** [comment]
yes, when the new projects are created, an email is sent out to this audit email address.

**Comment by pdeuter** [comment]
No, we should not use that address.  We need two distinct email addresses, otherwise all those emails will come to me and that is not what we want.

File: app/mailers/deploy_mailer.rb [hunk 1]
```diff
@@ -17,7 +17,8 @@ def bypass_email(stage, deploy, user)
     to_email = [BuddyCheck.bypass_email_address]
     to_email << BuddyCheck.bypass_jira_email_address if BuddyCheck.bypass_jira_email_address
 
-    mail(to: to_email, cc: user.email, subject: subject)
+    cc_email = [user.email, BuddyCheck.bypass_retroactive_approval_email]
```
Comment (jonmoter) [inline app/mailers/deploy_mailer.rb]: Maybe add a `compact` to the end of this, for non-zendesk Samson users that don't have that environment variable defined.

**Comment by jonmoter** [comment]
Are you going to open up a PR to chef_zendesk_samson to set the environment variable?  You'd probably want to set it around here:

https://github.com/zendesk/chef_zendesk_samson/blob/master/attributes/default.rb#L35

File: app/mailers/deploy_mailer.rb (hunk 1 above)
Comment (jwswj) [inline app/mailers/deploy_mailer.rb]:
```ruby
cc_email = [user.email]
cc_email << BuddyCheck.bypass_retroactive_approval_email if BuddyCheck.enabled?
```
or 
```ruby
cc_email = [user.email]
cc_email << BuddyCheck.bypass_retroactive_approval_email if BuddyCheck.bypass_retroactive_approval_email
```

Comment (jwswj) [inline app/mailers/deploy_mailer.rb]: We are the only people I know who use the Buddy functionality, so it's nice to keep it separate.

**Comment by apanzerj** [comment]
left this dangling...

**Pull request closed without merging**